crates/oxc_language_server/src/main.rs

```diff
@@ -240,34 +240,92 @@ impl LanguageServer for Backend {
-                        || "Fix this problem".into(),
-                        |s| format!("Fix this {s} problem"),
-                    );
-
+                let rule_name: Option<String> = None;
                 let fixed_content = report.fixed_content.clone().unwrap();
 
-                return Ok(Some(vec![CodeActionOrCommand::CodeAction(CodeAction {
-                    title,
-                    kind: Some(CodeActionKind::QUICKFIX),
-                    is_preferred: Some(true),
-                    edit: Some(WorkspaceEdit {
-                        #[expect(clippy::disallowed_types)]
-                        changes: Some(std::collections::HashMap::from([(
-                            uri,
-                            vec![TextEdit {
-                                range: fixed_content.range,
-                                new_text: fixed_content.code,
-                            }],
-                        )])),
-                        ..WorkspaceEdit::default()
+                return Ok(Some(vec![
+                    CodeActionOrCommand::CodeAction(CodeAction {
+                        title: report.diagnostic.message.split(':').next().map_or_else(
+                            || "Fix this problem".into(),
+                            |s| format!("Fix this {s} problem"),
+                        ),
+                        kind: Some(CodeActionKind::QUICKFIX),
+                        is_preferred: Some(true),
+                        edit: Some(WorkspaceEdit {
+                            #[expect(clippy::disallowed_types)]
+                            changes: Some(std::collections::HashMap::from([(
+                                uri.clone(),
+                                vec![TextEdit {
+                                    range: fixed_content.range,
+                                    new_text: fixed_content.code,
+                                }],
+                            )])),
+                            ..WorkspaceEdit::default()
+                        }),
+                        disabled: None,
+                        data: None,
+                        diagnostics: None,
+                        command: None,
+                    }),
+                    CodeActionOrCommand::CodeAction(CodeAction {
+                        title: rule_name.clone().map_or_else(
+                            || "Disable oxlint for this line".into(),
+                            |s| format!("Disable {s} for this line"),
+                        ),
+                        kind: Some(CodeActionKind::QUICKFIX),
+                        is_preferred: Some(false),
+                        edit: Some(WorkspaceEdit {
+                            #[expect(clippy::disallowed_types)]
+                            changes: Some(std::collections::HashMap::from([(
+                                uri.clone(),
+                                vec![TextEdit {
+                                    range: Range {
```

Comment:
If there is already an existing `eslint-disable-next-line` comment, then it should probably be updated instead of adding a completely new line.  Otherwise you end up with something that doesn't work.

This is what would be created today.
```
// eslint-disable-next-line some-rule
// eslint-disable-next-line no-var
var something = 'abc';
```

I believe this is what we need to create (untested).
```
// eslint-disable-next-line some-rule no-var
var something = 'abc';
```